node/pkg/evm/connectors/poller.go

```diff
@@ -0,0 +1,218 @@
+package connectors
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	ethEvent "github.com/ethereum/go-ethereum/event"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	ethHexUtils "github.com/ethereum/go-ethereum/common/hexutil"
+	"go.uber.org/zap"
+)
+
+type PollFinalizer interface {
+	IsBlockFinalized(ctx context.Context, block *NewBlock) (bool, error)
+}
+
+// BlockPollConnector polls for new blocks instead of subscribing when using SubscribeForBlocks. It allows to specify a
+// finalizer which will be used to only return finalized blocks on subscriptions.
+type BlockPollConnector struct {
+	Connector
+	DelayInMs           int
+	isEthPoS            bool
+	hasEthSwitchedToPoS bool
+	finalizer           PollFinalizer
+
+	blockFeed ethEvent.Feed
+	errFeed   ethEvent.Feed
+}
+
+func NewBlockPollConnector(ctx context.Context, baseConnector Connector, finalizer PollFinalizer, delayInMs int, isEthPoS bool) (*BlockPollConnector, error) {
+	connector := &BlockPollConnector{
+		Connector:           baseConnector,
+		DelayInMs:           delayInMs,
+		isEthPoS:            isEthPoS,
+		hasEthSwitchedToPoS: false,
+		finalizer:           finalizer,
+	}
+	err := supervisor.Run(ctx, "blockPoller", connector.run)
+	if err != nil {
+		return nil, err
+	}
+	return connector, nil
+}
+
+func (b *BlockPollConnector) run(ctx context.Context) error {
+	logger := supervisor.Logger(ctx).With(zap.String("eth_network", b.Connector.NetworkName()))
+
+	latestBlock, err := b.getBlock(ctx, logger, nil)
+	if err != nil {
+		return err
+	}
+	currentBlockNumber := new(big.Int).Set(latestBlock.Number)
+
+	timer := time.NewTimer(time.Millisecond) // Start immediately.
+	supervisor.Signal(ctx, supervisor.SignalHealthy)
+
+	var errorCount int
+	for {
+		select {
+		case <-ctx.Done():
+			timer.Stop()
+			return ctx.Err()
+		case <-timer.C:
+			var errorOccurred bool
```

Comment:
readability nit: maybe refactor this case block into a separate named function

Reply:
Did that and cleaned up the logic there